BUMP_VERSION.asciidoc

```diff
@@ -0,0 +1,221 @@
+= How to bump the Elasticsearch version
+
+[partintro]
+
+This is a short guide on all the steps required to bump the Elasticsearch
+version number to the next version after a release has been cut.
+
+== Introduction
+
+In order to make the steps concrete, it helps to work through an example.  
+In this scenario, suppose our current major release version of Elasticsearch
+is version 5, and Elasticsearch's current version of 5.x is 5.2.0.  The 
+next major release version is 6, which is reflected in the master branch.  
+
+== Steps to take for feature freeze
+
+When ready to feature freeze 5.2.0, the following steps must be taken.
+
+=== Create the branch
+
+Create a new `5.2` branch from the `5.x` branch and push it to Github.  
+The `5.2` branch will represent the 5.2 line of 5.x.  Note that `origin`
+refers to the remote setup to point to the Elastic organization's GitHub
+Elasticsearch repository (i.e., `git@github.com:elastic/elasticsearch.git`).
+
+[source,sh]
+-------------------------------------
+git fetch origin
+git checkout -b 5.2 origin/5.x
+git branch --set-upstream-to=origin/5.2
+git push origin
+-------------------------------------
+
+=== Update the version on the branch
+
+Update the version of the `5.x` branch to 5.3.0.  This requires the following
+steps:
+
+- Checkout a new branch for this work, from which you will issue a pull request against `5.x`:
+
+[source,sh]
+-------------------------------------
+  git checkout -b bump_version_to_5.3 origin/5.x
+-------------------------------------
+
+- In `core/src/main/java/org/elasticsearch/Version.java`, add a constant for 5.3.0,
+mimicking how the constants are created for other versions.  Set the `CURRENT` constant
+to the newly created 5.3.0 version constant.  Make sure the Lucene version is correct
+for the newly created version.
+
+[source,java]
+-------------------------------------
+  + public static final int V_5_3_0_ID_UNRELEASED = 5030099;
+  + public static final Version V_5_3_0_UNRELEASED = new Version(V_5_3_0_ID, org.apache.lucene.util.Version.LUCENE_6_4_0);
+  + public static final Version CURRENT = V_5_3_0_UNRELEASED;
+-------------------------------------
+
+- Update the `Version#fromId` method to include the new 5.3.0 version:
+
+[source,java]
+-------------------------------------
+   public static Version fromId(int id) {
+       switch (id) {
+  +       case V_5_3_0_ID_UNRELEASED:
+  +           return V_5_3_0_UNRELEASED;
+-------------------------------------
+
+- Update the elasticsearch version in `buildSrc/version.properties` to 5.3.0:
+
+[source,ini]
+-------------------------------------
+  -elasticsearch     = 5.2.0
+  +elasticsearch     = 5.3.0
+-------------------------------------
+
+- Update the version number in each of the `docs/*/index.asciidoc` files that
```

Comment:
I've made the changes I mentioned to the docs and centralised all the version attributes for all docs in the Elasticsearch repo in a single page: `docs/Versions.asciidoc`  (see https://github.com/elastic/elasticsearch/blob/master/docs/Versions.asciidoc)

Rather than using the perl script to update it, I'd recommend reviewing it manually and updating as appropriate.